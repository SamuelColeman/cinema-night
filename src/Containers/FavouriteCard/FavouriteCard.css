.favourites_card {
	border: 1px solid black;
	text-align: center;
}

.favourites__card--title {
	color: white;
  font-size: 6vh;
}

.favourites__card--img {
	width: 50%;
}