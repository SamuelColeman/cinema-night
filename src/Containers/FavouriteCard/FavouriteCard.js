import React from 'react';
import './FavouriteCard.css';
import { connect } from 'react-redux';
import fave from '../../images/transparent-heard.png'

export const FavouriteCard = ({ handleFavourite, movie, hasError }) => {
        return (
            <section className='favourites_card' id={movie.poster_path}>
                <h1 className='favourites__card--title'>{ movie.title }</h1>
                <img className="favourites__card--img" src={movie.poster_path} alt={movie.title} />
                <button onClick={() => handleFavourite( movie )}>Favorite</button>
                <h1>{ hasError }</h1>
            </section>
        )
}  

export const mapStateToProps = (state) => ({
    hasError: state.hasError,
})

export default connect(mapStateToProps)(FavouriteCard);