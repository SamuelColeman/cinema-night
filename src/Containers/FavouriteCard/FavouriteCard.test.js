import React from 'react';
import { shallow } from 'enzyme';
import { FavouriteCard, mapStateToProps } from './FavouriteCard';

    describe('FavouriteCard', () => {
        let wrapper;
        let mockHandleFavourite = jest.fn();
        let mockSelectedMovie = {
          "poster_path": "/gibberish.jpg",
          "movie_id": 222222,
          "title": "A title",
          "vote_average": 11,
          "overview": "It's cool.",
          "release_date": "2019-10-18"
        }
        let mockError ='Error'

        beforeEach(() => {
          wrapper = shallow(<FavouriteCard 
            key={mockSelectedMovie.movie_id}
            movie={mockSelectedMovie}
            handleFavourite={mockHandleFavourite}
            hasError={mockError}
          />)
        })
    
        it('should match snapshot with all data passing through correctly', () => {
            expect(wrapper).toMatchSnapshot();
  })

})
        