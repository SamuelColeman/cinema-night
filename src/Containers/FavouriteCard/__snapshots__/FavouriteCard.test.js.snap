// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteCard should match snapshot with all data passing through correctly 1`] = `undefined`;
